packages/file-search/src/node/file-search-service-impl.ts

```diff
@@ -32,11 +32,13 @@ export class FileSearchServiceImpl implements FileSearchService {
             ],
             ...options
         };
-        const args: string[] = [
+        let args: string[] = [
             '--files',
             '--sort-files',
-            '-u',
         ];
+        if (options.useGitIgnore) {
```

Comment:
The condition here is reversed (though I agree it's confusing).  To use the .gitignore file (`useGitIgnore == true`) means that we want to ignore the files listed in it, so we don't want to pass -u.  To not use the .gitignore file (`useGitIgnore` == false) means that we want to see the files listed in it, so we want to pass -u.

The code works because the condition is also reversed in the client.

For reference, here is the documentation for the `-u` option of ripgrep:

```
    -u, --unrestricted                      
            Reduce the level of "smart" searching. A single -u won't respect .gitignore
            (etc.) files. Two -u flags will additionally search hidden files and
            directories. Three -u flags will additionally search binary files.
            
            -uu is roughly equivalent to grep -r and -uuu is roughly equivalent to grep -a
            -r.
```

Reply:
reverse the condition, so when "gitIgnored === false", we add the flag "-u" to list the files filtered out by the .gitignored